collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
```

Comment:
Actually I just realized that doing this descriptor thing prohibits use from using slots, as getattr calls through to the descriptor...as long as we are ok losing this feature (slots) or changing it later if we decide its worth while, I will proceed with the other way.